### ISSUE TITLE
Revert to upstream LMS

### ISSUE THREAD

**Issue by alexeyr** [scalan/scalan#120]
Ideally we don't want to maintain a fork of LMS. Disentangle our changes and split them: contribute upstream, move to Scalan project, or remove. Assigning to @yzotov for now, @alexanderfilippov and @aslesarenko also had some changes. IIRC the main blocker is that we use both `ArrayOps` and `ArrayLoops`, and there is some conflict between them. Should probably be done after #119.

**Comment by alexeyr** [comment]
Another issue I remembered is that we have https://github.com/scalan/virtualization-lms-core/blob/scalan-develop/src/internal/GenericCodegen.scala#L251, which can be fixed by overriding in Scalan's BaseCodegen.
